src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs

```diff
@@ -1088,6 +1082,278 @@ await using (var context = InMemoryApplicationDbContext(contextId))
             MockUtils.VerifyAllMocks(contentService, releaseFileService);
         }
 
+        [Fact]
+        public async Task UpdateReleaseStatus()
+        {
+            var releaseId = Guid.NewGuid();
+            var adHocReleaseType = new ReleaseType {Title = "Ad Hoc"};
+            var release = new Release
+            {
+                Id = releaseId,
+                Type = adHocReleaseType,
+                Publication = new Publication {Title = "Old publication"},
+                ReleaseName = "2030",
+                TimePeriodCoverage = TimeIdentifier.March,
+                Slug = "2030-march",
+                PublishScheduled = DateTime.UtcNow,
+                NextReleaseDate = new PartialDate {Day = "15", Month = "6", Year = "2039"},
+                PreReleaseAccessList = "Access list",
+                Version = 0,
+                PreviousVersionId = releaseId
+            };
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                await context.AddRangeAsync(adHocReleaseType, release);
+                await context.SaveChangesAsync();
+            }
+
+            var contentService = new Mock<IContentService>(MockBehavior.Strict);
+            var releaseFileService = new Mock<IReleaseFileService>(MockBehavior.Strict);
+
+            contentService.Setup(mock =>
+                    mock.GetContentBlocks<HtmlBlock>(release.Id))
+                .ReturnsAsync(new List<HtmlBlock>());
+
+            var nextReleaseDateEdited = new PartialDate {Day = "1", Month = "1", Year = "2040"};
+
+            await using (var context = InMemoryApplicationDbContext(contextId))
+            {
+                var releaseService = BuildReleaseService(contentDbContext: context,
+                    contentService: contentService.Object,
+                    releaseFileService: releaseFileService.Object);
+
+                var result = await releaseService
+                    .UpdateReleaseStatus(
+                        releaseId,
+                        new ReleaseStatusUpdateViewModel
+                        {
+                            PublishMethod = PublishMethod.Scheduled,
+                            PublishScheduled = "2051-06-30",
+                            NextReleaseDate = nextReleaseDateEdited,
+                            ApprovalStatus = ReleaseApprovalStatus.Draft,
+                            LatestInternalReleaseNote = "Test internal note"
+                        }
+                    );
+
+                Assert.True(result.IsRight);
```

Comment:
You can use:

```
var updatedRelease = result.AssertRight();
```

instead for brevity.

Reply:
Then you also don't need to use `result.Right.` all the time below either

Reply:
I think this is only on the SOW branch at the moment?

Reply:
Oh sorry!